charts/datadog/templates/cluster-agent-rbac.yaml

```diff
@@ -233,7 +233,7 @@ metadata:
     app.kubernetes.io/instance: {{ .Release.Name | quote }}
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
     app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
-  name: {{ template "datadog.fullname" . }}-cluster-agent:system:auth-delegator
+  name: {{ template "datadog.fullname" . }}-cluster-agent-system-auth-delegator
```

Comment:
You shouldn't arbitrarily rename kubernetes resources like this.

Without an easy way to clean things up for non-helm users, this leaves 2 CRBs floating around in clusters.. Also there was no note of this change, and this repo has no changelog. it's not a great experience for datadog users